### ISSUE TITLE
Multiple Infinite Scroll ListViews On Single Page Creates Error with Refresh Database Request

### ISSUE THREAD

**Issue by EmersonFJ** [FlutterFlow/flutterflow-issues#1362]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

When a page has two ListView with Infinite Scroll enabled, you cannot have actions for Request Database Refresh for both ListViews. Just by creating the two actions, the application with fail on load. This error seems to be based on the fact that a "waitforOnePage" variable is declared with Infinite Scroll; however, I am not sure why having two actions, one each referencing a different ListView, for Request Database Refresh causes this error.

The error is as follows:

`Launching lib/main.dart on Web Server in debug mode...
Waiting for connection from debug service on Web Server...      
lib/pages/home_page/home_page_model.dart:130:10: Error: 'waitForOnePage' is already declared in this scope.
  Future waitForOnePage({
         ^^^^^^^^^^^^^^
lib/pages/home_page/home_page_model.dart:114:10: Context: Previous declaration of 'waitForOnePage'.
  Future waitForOnePage({
         ^^^^^^^^^^^^^^
Waiting for connection from debug service on Web Server...         43.2s
Failed to compile application.`

### Expected Behavior

I would expect that because it is now possible to have two or more ListViews with Infinite Scroll enabled on a single page (this was not previously possible; this is a huge benefit to me, so thank you), you should also be able to refresh those ListViews as well (this is the main source of benefit, however, for my use cases - haha - so hopefully this will be possible).

### Steps to Reproduce

1. Create a Page
2. Add two ListViews to the page, each with an API call, and Infinite Scroll enabled
3. Add two Buttons to the page, each with a Request Database Refresh action referencing the ListViews
4. Run Test Mode, and it will crash with the error above

See pre-built example: https://app.flutterflow.io/project/multiple-list-view-refresh-example-xer165

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT8Ki8/l8c9Nrs9D7q6JasJagmA5KVA/U40eluxEdxUaIozkP513Y8jRQFZvTf2+YmNmJWCkjTkextHtiYbpCvkCCwutbL5t+JNISBf/f3O6MMyDGbu0Q0Z8GeZXCna63KevsxJ+Ju1sdXNk7F2uBPaSXgjXQozROlUNQcj/GKjAo2ihAA/eVxd70TMORHrSONsqdA6ypIiUtpJh0eDnyw==

### Context

I am trying to have multiple ListViews on a single page that can be refreshed. This has prevented me from being able to do that. The use case here is having a social news feed that you can click to have "Most Popular", "Most Recent", "Your Friends Only" - which requires three list views with infinite scroll given the way the Backend Queries cannot be changed in a ListView. I then use conditional visibility to show the appropriate social news feed. 

The example I have linked to in here is a much simpler example of what I am doing, so hopeful it is helpful. Thank you!

### Visual documentation

Here is a link to a working example that creates the error; the two buttons hold the actions for Refresh Database Request. By changing the ListViews to no longer be Infinite Scroll-enabled, the error mentioned above goes away.

Link (also shared above): https://app.flutterflow.io/project/multiple-list-view-refresh-example-xer165

### Additional Info

Thank you for the help. Love FF - very close to launching and this is one of two gating items for me. 

### Environment

```markdown
- FlutterFlow version: 3.1
- Platform: Web
- Browser name and version: Chrome 114.0.5735.133
- Operating system and version affected:  MacOS
```

**Comment by AlexFox63** [comment]
Hey! Right now FlutterFlow has a workaround: you can convert your widget with 2 listviews to 2 different components with these listviews and it will work.

**Comment by EmersonFJ** [comment]
This is partially correct, at least I as I understand it. Please correct me if I am wrong.This is what I did back when you could not have two ListView with Infinite Scroll. However, the issue becomes you cannot call Refresh Database Request from a Page for a ListView in a component - you cannot access it. Again if you know how to do this let me know. So, if I am righty still, you would have to recreate your entire page as a custom component depending on your format and where your ideal “refresh” button place was…. If the page was complex, that becomes a huge challenge and creates other problems with trying to pass parameters and you would run into the same issue with actions being difficult depending on complexity.So I would say this is not a workout reallyThank you.Sent from my iPhoneOn Aug 26, 2023, at 17:30, Aleksandr Fokin ***@***.***> wrote:﻿
Hey! Right now FlutterFlow has a workaround: you can convert your widget with 2 listviews to 2 different components with these listviews and it will work.

**Comment by EmersonFJ** [comment]
It would be interesting if there was a response

**Comment by joshua-ffio** [comment]
Thank you for bringing this up @EmersonFJ . I'm happy to say that this issue has been fixed and I can confirm that the previously duplicated methods named `waitForOnePage()` are now uniquely named in the sample project that you provided. You should be able to see this fix live in FlutterFlow v4.0!

**Comment by EmersonFJ** [comment]
I confirm this is working on my end as well; thank you for pushing this one across the finish line. You guys are awesome!
